.github/workflows/ci.yml

```diff
@@ -338,82 +315,139 @@ jobs:
+        id: run
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        shell: bash
+        run: |
+          # ID for the latest successful run of the workflow on master push
+          # This is the run where test durations will have been uploaded
+          id=$(gh run list \
+            --workflow ${{ github.workflow }} \
+            --status=success \
+            --limit 1 \
+            --json databaseId \
+            --jq '.[].databaseId' \
+            --event push \
+            --branch master
+          )
+          echo "id:${id}"
+          echo "id=$id" >> $GITHUB_OUTPUT
+
+      - name: Calculate splits
+        id: result
+        shell: bash
+        run: |
+          splits=${{ inputs.splits || 14 }}
+          echo "splits: $splits"
+          echo "splits=$splits" >> $GITHUB_OUTPUT
 
-  test_internal_routes_allowed:
+          # Construct the matrix input for test_main using the groups count
+          # the matrix.group should be an array of numbers from 1 to $splits
+          matrix=[$(seq -s, 1 $splits)]
+          echo "matrix: $matrix"
+          echo "matrix=$matrix" >> $GITHUB_OUTPUT
+
+  test_main:
     runs-on: ubuntu-latest
-    needs: build
+    needs: [build, test_config]
+    strategy:
+      fail-fast: false
+      matrix:
+        group: ${{fromJson(needs.test_config.outputs.matrix)}}
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Test (test_internal_routes_allowed)
+      - name: Test (test_matrix)
         uses: ./.github/actions/run-docker
         with:
           services: ''
           digest: ${{ needs.build.outputs.digest }}
           version: ${{ needs.build.outputs.version }}
+          compose_file: docker-compose.yml
           run: |
-            make test_internal_routes_allowed
+            split="--splits ${{ needs.test_config.outputs.splits }}"
+            group="--group ${{ matrix.group }}"
+            report="--report-log ${{ env.log_file}}"
+            make test_main ARGS="${split} ${group} ${report}"
 
-  test_es_tests:
+      - name: Upload logs
+        uses: actions/upload-artifact@v4
+        with:
+          path: ${{ env.log_file }}
+          name: ${{ env.log_artifact }}-${{ matrix.group }}
+          retention-days: 1
+          overwrite: true
+
+  test_log:
     runs-on: ubuntu-latest
-    needs: build
+    if: always()
+    needs: [build, test_config, test_main]
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Test (test_es_tests)
-        uses: ./.github/actions/run-docker
+      - uses: actions/download-artifact@v4
         with:
-          services: ''
-          digest: ${{ needs.build.outputs.digest }}
-          version: ${{ needs.build.outputs.version }}
-          run: |
-            make test_es_tests
+          pattern: ${{ env.log_artifact }}*
+
+      - name: Cat logs
+        shell: bash
+        run: |
+          for dir in ${{ env.log_artifact }}*/; do
```

Comment:
Minor detail: the ordering here is confusing, it will do 1 10 11 12 13 14 2 3 4 5 6 7 8 9...

Reply:
It is doing alphabetical order as that is the order in the filesystem, is it important or relevant that it goes in numerical order?

Reply:
It's a bit confusing that the tests are displayed in numerical order in the jobs list but then displayed in alphabetical order in that logs summary we're building

Reply:
`sort -V` can handle that FYI

Reply:
Got it. Let's see if that works.